src/features/todos.ts

```diff
@@ -1,8 +1,17 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction, Slice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
+import { RootState } from '../app/store';
 
-export const todosSlice = createSlice({
+export const todosSlice: Slice<Todo[]> = createSlice({
   name: 'todos',
   initialState: [] as Todo[],
-  reducers: {},
+  reducers: {
+    setTodos(todos, action: PayloadAction<Todo[]>) {
+      todos.push(...action.payload);
```

Comment:
The `setTodos` reducer currently uses `push` to add todos to the state, which mutates the state directly. Instead, you should return the new array of todos directly: `return action.payload;`.